dbcrossbarlib/src/drivers/bigquery_shared/column.rs

```diff
@@ -62,24 +62,60 @@ impl BqColumn {
         f: &mut dyn Write,
         idx: usize,
     ) -> Result<()> {
-        if let BqDataType::Array(elem_ty) = &self.ty {
-            write!(
-                f,
-                r#"CREATE TEMP FUNCTION ImportJson_{idx}(input STRING)
+        match &self.ty {
+            // JavaScript UDFs can't return `DATETIME` yet, so we need a fairly
+            // elaborate workaround.
+            BqDataType::Array(BqNonArrayDataType::Datetime) => {
```

Comment:
BigQuery imports needed to get more complicated, mostly to import arrays. Arrays are kind of terrible, and I already regret trying to support them in v1. (It cost me a day or two.) But they do work.